File: 2 Even Fibonacci numbers/2 Even Fibonacci numbers.go
package main

//This program's C++ equivalent runs faster

import "fmt"

func main() {
	var T int
	fmt.Scanf("%d\n", &T)
	for i := 0; i < T; i++ {
		var n int64
		fmt.Scanf("%d\n", &n)
		fmt.Println(sumOfEvenFibonnaci(n))
	}
}

func sumOfEvenFibonnaci(n int64) int64 {
	var a, b int64 = 2, 8
	var sum int64 = 2
	for b <= n {
		sum += b
		a, b = b, 4*b+a
	}
	return sum
}

File: 2 Even Fibonacci numbers/2 Even Fibonacci numbers.cpp
#include <iostream>
using namespace std;

long sumOfEvenFibonnaci(long n){
    long a =2, b = 8;
    long sum = 2;
    while (b<=n){
        sum +=b;
        long tmp = a;
        a = b;
        b = 4*b + tmp;
    }
    return sum;
}

int main(){
    int T;
    cin >> T;
    for (size_t i = 0; i < T; i++){
        long n;
        cin >> n;
        cout << sumOfEvenFibonnaci(n)<<endl;
    }
    return 0;
}